osu.Game/Rulesets/Mods/ModBarrelRoll.cs

```diff
@@ -50,8 +50,8 @@ public void ApplyToDrawableRuleset(DrawableRuleset<TObject> drawableRuleset)
 
             var playfieldSize = drawableRuleset.Playfield.DrawSize;
             var minSide = MathF.Min(playfieldSize.X, playfieldSize.Y);
-            var maxSide = MathF.Max(playfieldSize.X, playfieldSize.Y);
-            drawableRuleset.Playfield.Scale = new Vector2(minSide / maxSide);
+            var diagonal = MathF.Sqrt(MathF.Pow(playfieldSize.X, 2) + MathF.Pow(playfieldSize.Y, 2));
+            drawableRuleset.Playfield.Scale = new Vector2(minSide / diagonal);
```

Comment:
```suggestion
            drawableRuleset.Playfield.Scale = new Vector2(minSide / playfieldSize.LengthFast);
```

Reply:
Not sure why go for `LengthFast`, it's not exactly a hot path - this gets calculated and set once, so may as well do normal `Length` - but whichever works I guess...?

Reply:
It's randomly used across the game which confused me, either way works anyways.